src/caustics/lenses/multipole.py

```diff
@@ -0,0 +1,228 @@
+        The redshift of the lens.
+    x0, y0: Optional[Union[Tensor, float]]
+        Coordinates of the shear center in the lens plane.
+    a_m: Optional[Union[Tensor, float]]
+        Strength of multipole.
+    phi_m: Optional[Union[Tensor, float]]
+        Orientation of multiple.
+    m: Optional[Union[Tensor, int]]
+        Order of multipole.
+
+    """
+
+    _null_params = {"x0": 0.0, "y0": 0.0, "a_m": 0.1, "phi_m": 0.0, "m": 3}
+
+    def __init__(
+        self,
+        cosmology: CosmologyType,
+        z_l: ZLType = None,
+        x0: Annotated[
+            Optional[Union[Tensor, float]], "The x-coordinate of the lens center", True
+        ] = None,
+        y0: Annotated[
+            Optional[Union[Tensor, float]], "The y-coordinate of the lens center", True
+        ] = None,
+        a_m: Annotated[
+            Optional[Union[Tensor, float]], "The amplitude of the multipole", True
+        ] = None,  # TODO change to true axis ratio
+        phi_m: Annotated[
+            Optional[Union[Tensor, float]],
+            "The orientation angle of the multipole",
+            True,
+        ] = None,
+        m: Annotated[
+            Optional[Union[Tensor, int]], "The Multipole moment m", True
+        ] = None,
+        name: NameType = None,
+    ):
+        super().__init__(cosmology, z_l, name=name)
+
+        self.add_param("x0", x0)
+        self.add_param("y0", y0)
+        self.add_param("a_m", a_m)
+        self.add_param("phi_m", phi_m)
+        if type(m) is not type(None) and m < 2:
+            raise ValueError("Multipole order must be greater or equal to 2")
+        self.add_param("m", m)
+
+    @unpack
+    def reduced_deflection_angle(
+        self,
+        x: Tensor,
+        y: Tensor,
+        z_s: Tensor,
+        *args,
+        params: Optional["Packed"] = None,
+        z_l: Optional[Tensor] = None,
+        x0: Optional[Tensor] = None,
+        y0: Optional[Tensor] = None,
+        a_m: Optional[Tensor] = None,
+        phi_m: Optional[Tensor] = None,
+        m: Optional[Tensor] = None,
+        **kwargs,
+    ) -> tuple[Tensor, Tensor]:
+        """
+        Calculate the deflection angle of the multipole.
+
+        Parameters
+        ----------
+        x: Tensor
+            The x-coordinate of the lens.
+
+            *Unit: arcsec*
+
+        y: Tensor
+            The y-coordinate of the lens.
+
+            *Unit: arcsec*
+
+        z_s: Tensor
+            The source redshift.
+
+            *Unit: unitless*
+
+        params: (Packed, optional)
+            Dynamic parameter container.
+
+        Returns
+        -------
+        x_component: Tensor
+            Deflection Angle
+
+            *Unit: arcsec*
+
+        y_component: Tensor
+            Deflection Angle
+
+            *Unit: arcsec*
+
+        Equation (B11) and (B12) https://arxiv.org/pdf/1307.4220, Xu et al. 2014
+
+        """
+        return func.reduced_deflection_angle_multipole(x0, y0, m, a_m, phi_m, x, y)
```

Comment:
Could you make this go over a loop in the `self.m` values and add the contributions? Could even vmap it.